src/v2/utils/parser/ScheduleParser.ts

```diff
@@ -203,28 +204,55 @@ export class ScheduleParser implements Parser {
             disciplineTypeId: disciplineType.id,
           },
         },
+      });
+    } else if (
+      event.startTime.getTime() === startOfEvent.getTime() - WEEK &&
+        event.period === Period.EVERY_FORTNIGHT
+    ) {
+      await this.eventRepository.updateById(event.id, {
+        period: Period.EVERY_WEEK,
+        endTime: endOfEvent,
+      });
+    }
+    await this.updateEventInfo(event, startOfEvent, endOfEvent);
+  }
+
+  async updateEventInfo (event: DbEvent, startOfEvent: Date, endOfEvent: Date) {
+    if (!event) {
+      await this.eventRepository.updateById(event.id, {
         eventInfo: {
           createMany: {
-            data: this.getIndexesForEventInfo(0, 8),
+            data: this.getIndexesForEventInfo(0, await this.getWeeksCount(endOfEvent, Period.EVERY_FORTNIGHT)),
```

Comment:
You had better create a variable `weeksCountFortnight`, because you used **getWeeksCount** method twice